src/pages/index.js

```diff
@@ -1,87 +1,237 @@
+  })
+  .catch((err) => {
+    console.log(err);
+  });
+
+function renderCard(cardData) {
+  const card = new Card(
+    cardData,
+    selectors.cardTemplate,
+    openPreviewModal,
+    handleCardDeleteClick,
+    handleLikeClick
+  ).getCardElement();
+  cardSection.addItem(card);
 }
 
-const userInformation = new UserInfo({
+const userInfo = new UserInfo({
   name: ".profile__title",
   description: ".profile__description",
+  userAvatarSelector: ".profile__avatar",
 });
 
-const profileEditForm = new PopupWithForm("#profile-edit-modal", (data) => {
-  console.log("Form data", data);
-  userInformation.setUserInfo({
-    title: data.title,
-    description: data.description,
+const profileEditForm = new PopupWithForm(
+  "#profile-edit-modal",
+  handleProfileEditSubmit
+);
+const addCardPopup = new PopupWithForm("#add-card-modal", handleCardFormSubmit);
+const previewImagePopup = new PopupWithImage("#modal__preview-card");
+const confirmDeletePopup = new PopupWithConfirm("#delete-card-modal");
+
+const avatarImagePopup = new PopupWithForm(
+  "#edit-avatar-modal",
+  handleAvatarFormSubmit
+);
+
+/*Validation*/
+
+const editFormValidator = new FormValidator(config, profileForm);
+const addCardValidator = new FormValidator(config, cardForm);
+const avatarValidator = new FormValidator(config, avatarUpdateForm);
+
+/*Event Handlers*/
+
+function handleProfileEditSubmit(inputItems) {
+  profileEditForm.renderLoading(true);
+  api
+    .userProfileInfo(inputItems.name, inputItems.about)
+    .then((res) => {
+      console.log(res);
+      userInfo.setUserInfo(res);
+      profileEditForm.close();
+    })
+    .catch((err) => {
+      console.log(err);
+    })
+    .finally(() => {
+      profileEditForm.renderLoading(false);
+    });
+}
+function handleCardFormSubmit(data) {
+  addCardPopup.renderLoading(true);
+  api
+    .addNewCard(data.name, data.link)
+    .then((res) => {
+      renderCard(res);
+      addCardPopup.close();
+      addCardValidator.resetValidation();
+    })
+    .catch((err) => {
+      console.log(err);
+    })
+    .finally(() => {
+      addCardPopup.renderLoading(false);
+    });
+}
+
+function handleLikeClick(card) {
+  console.log(card);
+  if (card.isLiked) {
+    api
+      .dislikeCard(card._id)
+      .then(() => {
+        card.handleLikeButton();
+        card.isLiked = false;
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  }
+  if (!card.isLiked) {
+    api
+      .likeCard(card._id)
+      .then(() => {
+        card.handleLikeButton();
+        card.isLiked = true;
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  }
+}
+
+function handleCardDeleteClick(card) {
+  console.log(card);
+  confirmDeletePopup.open();
+  confirmDeletePopup.handleDelete(() => {
+    api
+      .deleteCard(card._id)
+      .then(() => {
+        confirmDeletePopup.close();
+        card.handleTrashButton();
+      })
+      .catch((err) => {
+        console.log(err);
```

Comment:
Good job that you `catch` possible errors at the end of server requests. You can simplify it with `.catch(console.error);` because `console.error` is meant to log errors